### PR TITLE
WIP: Kernel log parser

### PR DESCRIPTION
**Pull request by chaws**
Closes https://github.com/Linaro/squadplugins/issues/3 and possibly https://github.com/Linaro/squad/issues/592

I managed to put several regex'es together so that `findall` is called only once. Not sure if that has better performance than running each regex on its own. Also, there's a pre-regex that filters-out all non-kernel messages (that don't start with `[number.number]`), thus reducing log size considerably (specially in jobs with lots of target output).

By default, there are 6 tests, each represented by a regex, all under **kernel-issues** suite:
* if a test has no match, result=True
* if a test has at least one match, result=False
* if a test matches more than once, all occurrences are concatenated and saved to its log. Meaning there will always be one test per build per testrun (2 Oopses == 1 test)

I'll put it to work with SQUAD tomorrow to see how it behaves, but I think we can discuss the behavior already.

PS: also fixed a minor issue regarding yaml.load

### CODE REVIEW

File: kernel_issues/__init__.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        for match in matches:
+            for regex_id in range(len(REGEXES)):
+                if len(match[regex_id]) > 0:
+                    snippets[regex_id].append(match[regex_id])
+
+        suite, _ = testrun.build.project.suites.get_or_create(slug='kernel-issues')
+        for regex_id in range(len(REGEXES)):
+            testrun.tests.create(
+                suite=suite,
+                name=REGEXES[regex_id][0],
```
Comment (danrue) [inline kernel_issues/__init__.py]: Can you give an example of what suite and name will resolve to?

**Comment by danrue** [comment]
This looks OK, but it is a bit hard to tell until we give it a try.

File: kernel_issues/__init__.py (hunk 1 above)
Comment (terceiro) [reply kernel_issues/__init__.py]: kernel-issues/kernel_panic
kernel-issues/kernel_exception
kernel-issues/kernel_trace
kernel-issues/kernel_oops
kernel-issues/kernel_fault
kernel-issues/kernel_warning

these names are defined at the top

Comment (danrue) [reply kernel_issues/__init__.py]: We run ~20 lava jobs per environment, and so this test result will then get overwritten 20 times and last one wins. The name needs to be per-job somehow.

**Comment by terceiro** [comment]
LGTM, but as we discussed earlier, this should instead be in squad since this type of parsing is not 
linaro-specific. the linux_log_parser plugin in the squad  tree does the same as this, and it should be improved instead of creating a separate implementation.

it would also be nice to have some benchmark comparing this implementation to the existing one in the squad tree (which of course is not optimized at all, and should probably lose)

**Comment by mwasilew** [comment]
I have quite opposite view - the plugin from squad should be removed and replaced with this one.

**Comment by chaws** [comment]
Moved PR to squad

**Pull request closed without merging**